src/simulator/layer.c

```diff
@@ -89,22 +181,41 @@ layer_step_inject_current(Layer* layer, u32 time, f32* currents, u32 n_currents)
 
 
 internal void
-layer_step_force_spike(Layer* layer, u32 time, bool* spikes, u32 n_spikes) {
+layer_step_force_spike(Layer* layer, u32 time, bool* spikes, u32 n_spikes,
+                       MemoryArena* storage, ThreadPool* pool) {
     TIMING_COUNTER_START(LAYER_STEP_FORCE_SPIKE);
     
     check(layer != NULL, "layer is NULL");
     check(spikes != NULL, "spikes is NULL");
+    check(storage != NULL, "storage is NULL");
+    check(pool != NULL, "pool is NULL");
+    
+    thread_pool_reset(pool);
     
     u32 n_inputs = math_min_u32(layer->n_neurons, n_spikes);
-    u32 i = 0;
-    for (i = 0; i < n_inputs; ++i) {
-        if (spikes[i] == TRUE) 
-            neuron_step_force_spike(layer->neurons + i, time);
-        else
-            neuron_step(layer->neurons + i, time);
+    
+    u32 n_tasks = pool->n_threads + 1; // consider also the main thread
+    u32 neurons_per_task = (layer->n_neurons + n_tasks - 1) / n_tasks;
+    
+    for (u32 task_i = 0; task_i < n_tasks; ++task_i) {
```

Comment:
A lot of code duplication between the 3 step functions, can we do better?

Reply:
Done, added some helper functions to reduce the code in the functions, they are still similar but not the critical code